ui/src/main/java/edu/wpi/grip/ui/codegeneration/TemplateMethods.java

```diff
@@ -0,0 +1,137 @@
+import edu.wpi.grip.generated.opencv_core.enumeration.CmpTypesEnum;
+import edu.wpi.grip.generated.opencv_core.enumeration.LineTypesEnum;
+import edu.wpi.grip.ui.codegeneration.data.TStep;
+
+import com.google.common.base.CaseFormat;
+
+import org.bytedeco.javacpp.opencv_core.Point;
+import org.bytedeco.javacpp.opencv_core.Scalar;
+import org.bytedeco.javacpp.opencv_core.Size;
+
+import java.nio.DoubleBuffer;
+import java.util.Optional;
+
+public abstract class TemplateMethods {
+
+  /**
+   * gets a TemplateMethod of the desired language.
+   *
+   * @param lang the desired language that will be exported
+   * @return The template method of the correct language.
+   */
+  public static TemplateMethods get(Language lang) {
+    switch (lang) {
+      case JAVA:
+        return new JavaTMethods();
+      case PYTHON:
+        return new PythonTMethods();
+      case CPP:
+        return new CppTMethods();
+      default:
+        throw new IllegalArgumentException(
+            lang + " is not a supported language for code generation.");
+    }
+  }
+
+  /**
+   * takes a socket and returns the value.
+   *
+   * @param socket the socket to be parsed
+   * @return the value of the socket or "null" if there is no value
+   */
+  public static String parseSocketValue(Socket socket) {
+    if (socket.getValue().isPresent() && !socket.getValue().get().toString().contains("bytedeco")
+        && !socket.getValue().get().toString().contains("Infinity")
+        && !socket.getValue().get().toString().contains("ContoursReport")) {
+      return socket.getValue().get().toString();
+    } else {
+      Optional initValOptional = socket.getSocketHint().createInitialValue();
+      if (initValOptional.isPresent()) {
+        Object initVal = initValOptional.get();
+        String type = parseSocketType(socket);
+        StringBuilder valueBuilder = new StringBuilder();
+        if ("Point".equalsIgnoreCase(type)) {
+          Point pointVal = (Point) initVal;
+          valueBuilder.append('(').append(pointVal.x()).append(", ").append(pointVal.y())
+              .append(')');
+        } else if ("Size".equalsIgnoreCase(type)) {
+          Size sizeVal = (Size) initVal;
+          valueBuilder.append('(').append(sizeVal.width()).append(", ").append(sizeVal.height())
+              .append(')');
+        } else if ("Scalar".equals(type)) {
+          Scalar scaleVal = (Scalar) initVal;
+          DoubleBuffer buff = scaleVal.asBuffer();
+          StringBuilder temp = new StringBuilder();
+          temp.append('(').append(buff.get());
+          while (buff.hasRemaining()) {
+            temp.append(", ").append(buff.get());
+          }
+          temp.append(')');
+          if (temp.toString().contains("E")) {
+            valueBuilder.append("(-1)");
+          } else {
+            valueBuilder.append(temp.toString());
+          }
+        } else if ("Mat".equals(type)) {
+          return "";
+        }
```

Comment:
Do you think this could be rewritten to dispatch on type instead of this if else block?